src/bgw/job.c

```diff
@@ -974,12 +974,21 @@ ts_bgw_job_check_max_retries(BgwJob *job)
 }
 
 void
-ts_bgw_job_permission_check(BgwJob *job)
+ts_bgw_job_permission_check(BgwJob *job, const char *cmd)
```

Comment:
this method `ts_bgw_job_permission_check` seems to be only called from `tsl/src/bgw_policy/job_api.c`.

I wonder why the dll export is needed...

same seems to be true for other methods as well in this `job.c` ; like `ts_bgw_job_validate_schedule_interval`

Reply:
> this method `ts_bgw_job_permission_check` seems to be only called from `tsl/src/bgw_policy/job_api.c`.
> 
> I wonder why the dll export is needed...
> 
> same seems to be true for other methods as well in this `job.c` ; like `ts_bgw_job_validate_schedule_interval`

Good question. I had to add a few dll exports to make the tests pass, but let's see if we need it for these.

Reply:
Looked into the code, and there are some more refactorings we can do, so merging this and we'll handle it as a separate PR.